flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentControllerTest.java

```diff
@@ -397,7 +397,45 @@ public void testUpgradeNotReadyCluster() {
         testUpgradeNotReadyCluster(appCluster, false);
     }
 
-    public void testUpgradeNotReadyCluster(FlinkDeployment appCluster, boolean allowUpgrade) {
+    @Test
+    public void verifyReconcileWithBadConfig() throws Exception {
```

Comment:
This test does not cover that observer should use previous validated config, not the current invalid one. Right?

Reply:
You are right. Current test only verifies the order of `observe()` and `validate()`.
I enhanced the test to verify the previous validated config is really used by modifying the rest port.